### PR TITLE
apply alpha value into RGB channel of each pixel

### PR DESCRIPTION
**Pull request by FtheGFW**
pre-multiply alpha value into rgb channel after decoding the webp image into a bitmap's raw bytes

### CODE REVIEW

File: webp-android-backport-library/jni/webpbackport/android_backport_webp_WebPFactory.cpp [hunk 1]
```diff
@@ -100,6 +100,21 @@ JNIEXPORT jobject JNICALL Java_android_backport_webp_WebPFactory_nativeDecodeByt
...
+  int end = bitmapInfo.height * bitmapInfo.stride - 4;
+  while(index < end){
+    alpha = pixelsAddr[index+3];
+    normalizedAlpha = alpha / 255.0;
+    pixelsAddr[index] = (uint8_t)(pixelsAddr[index]  * normalizedAlpha);
+    pixelsAddr[index+1] = (uint8_t)(pixelsAddr[index+1]  * normalizedAlpha);
+    pixelsAddr[index+2] = (uint8_t)(pixelsAddr[index+2]  * normalizedAlpha);
+    index += 4;
+  }
+
```
Comment (alexey-pelykh) [inline webp-android-backport-library/jni/webpbackport/android_backport_webp_WebPFactory.cpp]: Please use correct spacing

File: example-project/src/com/example/backport/webp/SampleActivity.java [hunk 2]
```diff
@@ -56,7 +56,7 @@ public void onCreate(Bundle savedInstanceState) {
 		final Button loadEmbeddedImageButton = (Button) findViewById(R.id.loadEmbeddedImage);
 		loadEmbeddedImageButton.setOnClickListener(new View.OnClickListener() {
 			public void onClick(View v) {
-				InputStream rawImageStream = getResources().openRawResource(R.raw.image);
+				InputStream rawImageStream = getResources().openRawResource(R.raw.alpha_image);
```
Comment (alexey-pelykh) [inline example-project/src/com/example/backport/webp/SampleActivity.java]: Please extend sample with extra feature, don't replace original

File: webp-android-backport-library/jni/webpbackport/android_backport_webp_WebPFactory.cpp (hunk 1 above)
Comment (FtheGFW) [inline webp-android-backport-library/jni/webpbackport/android_backport_webp_WebPFactory.cpp]: updated

**Comment by alexey-pelykh** [comment]
Please correct formatting and adjust the pull-request so I can merge it

**Comment by FtheGFW** [comment]
The following screen shots were captured from SourceTree displaying the code changes of the commit, as you can see, the format is well, so I think the format problem may be github's fault.   
screen shot 2016-07-13 at 11 07 40 am   
screen shot 2016-07-13 at 11 07 54 am

**Comment by FtheGFW** [comment]
And what do you mean "adjust the pull-request" ? change the pr's title ? sorry, I'm not a native English speaking guy, would you please be more specific, thanks!

**Comment by alexey-pelykh** [comment]
image
This file uses tabs instead of spaces, that's the only thing that needs to be adjusted.

**Comment by FtheGFW** [comment]
updated

**Pull request merged**